File: tasks/src/main/java/com/example/tasks/dto/TaskDto.java
package com.example.tasks.dto;

import com.example.tasks.entity.Status;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDate;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
public class TaskDto {

    private Long id;
    private String toDo;
    private Status status;
    private Long userId;
    private String username;
    private LocalDate beginDate;
    private LocalDate completedDate;
}

File: tasks-interface/src/app/dtos/TaskDto.ts
export class TaskDto {
  id?: number;
  toDo?: string;
  status?: string;
  userId?: number;
  beginDate?: string;
  completedDate?: string;


  constructor(id: number, toDo: string, status: string, user: number, beginDate: string, completedDate: string) {
    this.id = id;
    this.toDo = toDo;
    this.status = status;
    this.userId = user;
    this.beginDate = beginDate;
    this.completedDate = completedDate;
  }
}
